scripts/hud/customizer.ts

```diff
@@ -130,129 +126,148 @@ const snaps: Record<
+			component.panel.SetPanelEvent('onmouseover', () => this.onComponentMouseOver(component));
+		}
+	}
 
-	for (const [id, component] of Object.entries(components)) {
-		const posX = fixFloatingImprecision(
-			LayoutUtil.getX(component.panel) +
-				LayoutUtil.getWidth(component.panel) * snaps[0][component.snaps[0]].sizeFactor
-		);
-		const posY = fixFloatingImprecision(
-			LayoutUtil.getX(component.panel) +
-				LayoutUtil.getHeight(component.panel) * snaps[1][component.snaps[1]].sizeFactor
-		);
+	public disableEditing(): void {
+		this.panels.customizer.RemoveClass('hud-customizer--enabled');
+		for (const component of Object.values(this.components)) {
+			component.panel.RemoveClass('hud-customizable');
+			component.panel.ClearPanelEvent('onmouseover');
+		}
+		$.UnregisterEventHandler('DragStart', this.panels.virtual, this.dragStartHandle);
+	}
 
-		saveData.components[id] = {
-			position: [posX, posY],
-			snaps: component.snaps
+	private save(): void {
+		const saveData: HudConfig = {
+			settings: { gridSize: this.gridSize },
+			components: {}
 		};
-	}
 
-	HudCustomizerAPI.SaveLayoutFromJS(saveData);
-}
+		for (const [id, component] of Object.entries(this.components)) {
+			const posX = fixFloatingImprecision(
+				LayoutUtil.getX(component.panel) +
+					LayoutUtil.getWidth(component.panel) * snaps[Axis.X][component.snaps[Axis.X]].sizeFactor
+			);
+			const posY = fixFloatingImprecision(
+				LayoutUtil.getY(component.panel) +
+					LayoutUtil.getHeight(component.panel) * snaps[Axis.Y][component.snaps[Axis.Y]].sizeFactor
+			);
+			saveData.components[id] = {
+				position: [posX, posY],
+				snaps: component.snaps
+			};
+		}
 
-function load() {
-	gridlines = [[], []];
-	activeGridlines = [undefined, undefined];
-	activeComponent = undefined;
-
-	let layoutData: HudConfig;
-	try {
-		layoutData = HudCustomizerAPI.GetLayout() as HudConfig;
-	} catch {
-		$.Warning('HudCustomizer: Failed to parse layout file!');
-		return;
+		HudCustomizerAPI.SaveLayoutFromJS(saveData);
 	}
 
-	gridSize = layoutData?.settings?.gridSize ?? 5;
-	panels.gridSize.value = gridSize;
+	private load(): void {
+		this.gridlines = [[], []];
+		this.activeGridlines = [undefined, undefined];
+		this.activeComponent = undefined;
 
-	createGridLines();
+		let layoutData: HudConfig;
+		try {
+			layoutData = HudCustomizerAPI.GetLayout() as HudConfig;
+		} catch {
+			$.Warning('HudCustomizer: Failed to parse layout file!');
+			return;
+		}
 
-	resizeKnobs = {} as any;
-	Object.values(ResizeDirection)
-		.filter((x) => !Number.isNaN(+x))
-		.forEach((dir) => {
-			resizeKnobs[dir] = $.CreatePanel('Button', panels.knobContainer, `Resize_${ResizeDirection[dir]}`, {
-				class: 'hud-customizer__resize-knob',
-				draggable: true,
-				hittest: true
-			});
-		});
+		this.gridSize = layoutData?.settings?.gridSize ?? DEFAULT_GRID_SIZE;
+		this.panels.gridSize.value = this.gridSize;
+
+		this.createGridLines();
+
+		this.resizeKnobs = {} as any;
```

Comment:
`Enum.values` handles filtering non-numeric values, so you can get rid of this `.filter`.

Reply:
Addressed in 74f396aa55355f1288479bab2290b85024d24fa6